### PR TITLE
Consistent ConfigureAwait on Loopback, RemoteExecutor.DisposeAsync

### PR DESCRIPTION
**Pull request by liveans**
This PR is doing couple of stuff:
- Adding an async disposal to RemoteExecutor and using it in tests.
- Using `ConfigureAwait(false)` more consistently on Http3Loopback(Server/Connection/Stream) classes.

### CODE REVIEW

File: src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientTest.cs [hunk 1]
```diff
@@ -854,20 +854,20 @@ public void DefaultProxy_Get_ReturnsNotNull()
...
-            }).Dispose();
+            }).DisposeAsync();
         }
 
         [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
-        public void DefaultProxy_Credentials_SetGet_Roundtrips()
+        public async Task DefaultProxy_Credentials_SetGet_Roundtrips()
         {
-            RemoteExecutor.Invoke(() =>
+            await RemoteExecutor.Invoke(() =>
```
Comment (rzikm) [inline src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientTest.cs:870]: All these can simply be returns in non `async` methods.

Not an important factor here, but it avoids allocation of a state machine and other small overhead, so it si useful to keep this in mind for more sensitive applications.

**Comment by ManickaP** [comment]
We also have remote executor test in Quic and we should look into S.N.Security and S.N.Sockets as well (if not in this PR, then by filing issues).

And I still think we should also file an issue in dotnet/arcade to ask for proper `DisposeAsync` and shortly describe the issue here and what we discovered.

**Comment by liveans** [comment]
> We also have remote executor test in Quic and we should look into S.N.Security and S.N.Sockets as well (if not in this PR, then by filing issues).
> 
> And I still think we should also file an issue in dotnet/arcade to ask for proper `DisposeAsync` and shortly describe the issue here and what we discovered.

I'm thinking to cover all RemoteExecutors in this PR, if we're agreed on this approach. I'll also file an issue in dotnet/arcade.

File: src/libraries/System.Net.Sockets/tests/FunctionalTests/SocketDuplicationTests.cs [hunk 2]
```diff
@@ -324,17 +324,20 @@ public async Task DuplicateAndClose_TcpServerHandler(AddressFamily addressFamily
...
-                    RunCommonHostLogic(hServerProc.Process.Id);
+                    RemoteInvokeHandle hServerProc = RemoteExecutor.Invoke(HandlerServerCode, _ipcPipeName);
+                    await RunCommonHostLogic(hServerProc.Process.Id);
+                    await hServerProc.DisposeAsync();
                 }
 
-                void RunCommonHostLogic(int processId)
+                async Task RunCommonHostLogic(int processId)
                 {
+                    await Task.Yield();
```
Comment (MihaZupan) [inline src/libraries/System.Net.Sockets/tests/FunctionalTests/SocketDuplicationTests.cs:339]: If the intention here was to ditch the sync context for the `WaitForConnection` call, you'd have to `ConfigureAwait(false)` or use something like
```c#
await Task.CompletedTask.ConfigureAwait(ConfigureAwaitOptions.ForceYielding);
```

**Review by MihaZupan — APPROVED** [review]

File: src/libraries/Common/tests/System/Net/RemoteExecutorExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Tasks;
+using Microsoft.DotNet.RemoteExecutor;
+
+namespace Microsoft.DotNet.RemoteExecutor;
+
+internal static class RemoteExecutorExtensions
+{
+    public static Task DisposeAsync(this RemoteInvokeHandle handle) =>
```
Comment (antonfirsov) [inline src/libraries/Common/tests/System/Net/RemoteExecutorExtensions.cs:11]: Note: [`IAsyncDisposable.DisposeAsync`](https://learn.microsoft.com/en-us/dotnet/api/system.iasyncdisposable.disposeasync) returns a `ValueTask`. If we expect https://github.com/dotnet/arcade/issues/14807 to be addressed by implementing the interface, the signature will change.

File: src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.cs [hunk 4]
```diff
@@ -999,11 +998,6 @@ public async Task SendAsync_ReadFromSlowStreamingServer_PartialDataReturned()
         [ActiveIssue("https://github.com/dotnet/runtime/issues/65429", typeof(PlatformDetection), nameof(PlatformDetection.IsNodeJS))]
         public async Task ReadAsStreamAsync_HandlerProducesWellBehavedResponseStream(bool? chunked, bool enableWasmStreaming, bool slowChunks)
         {
-            if (UseVersion == HttpVersion30)
-            {
-                throw new SkipTestException("https://github.com/dotnet/runtime/issues/91757");
```
Comment (CarnaViire) [inline src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.cs:1004]: Can we keep the disabled tests as disables and parallelization off in this PR, and separate re-enabling stuff to a separate follow up PR?

Comment (ManickaP) [reply src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.cs:1004]: On top of that, PR that re-enables these test, should also close those issues so we don't have leftover issues for already fixed stuff.

**Comment by liveans** [comment]
@CarnaViire @ManickaP I'll merge this as soon as CI finishes to run (if there are no related failures ofc). Are there any further concerns with this PR? Any other CI run you want to see?
I'll make a follow-up PR to close necessary issues and to re-enable parallelism/disabled tests.

File: src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.AltSvc.cs [hunk 5]
```diff
@@ -81,11 +81,6 @@ fromVersion.Major switch
         [ConditionalFact]
         public async Task AltSvc_ConnectionFrame_UpgradeFrom20_Success()
         {
-            if (UseVersion == HttpVersion30)
```
Comment (ManickaP) [inline src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.AltSvc.cs:84]: Is this intentional? Is the test fixed?

**Review by ManickaP — APPROVED** [review]
LGTM module one comment.

File: src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.AltSvc.cs (hunk 5 above)
Comment (liveans) [reply src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.AltSvc.cs:84]: Ah nice catch, thanks! Nope, this is not intentional. Will fix it

**Review by CarnaViire — APPROVED** [review]
LGTM, thanks!

**Pull request merged**